exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata.java

```diff
@@ -675,32 +699,28 @@ private boolean tableModified(List<Path> directories, Path metaFilePath, Path pa
     FileStatus directoryStatus = fs.getFileStatus(parentDir);
     int numDirs = 1;
     if (directoryStatus.getModificationTime() > metaFileModifyTime) {
-      if (timer != null) {
-        logger.debug("Directory {} was modified. Took {} ms to check modification time of {} directories",
-            directoryStatus.getPath().toString(), timer.elapsed(TimeUnit.MILLISECONDS), numDirs);
-        timer.stop();
-      }
-      return true;
+      return logAndStopTimer(true, timer, numDirs);
     }
+    boolean isModified = false;
     for (Path directory : directories) {
       numDirs++;
       metaContext.setStatus(directory);
       directoryStatus = fs.getFileStatus(directory);
       if (directoryStatus.getModificationTime() > metaFileModifyTime) {
-        if (timer != null) {
-          logger.debug("Directory {} was modified. Took {} ms to check modification time of {} directories",
-              directoryStatus.getPath().toString(), timer.elapsed(TimeUnit.MILLISECONDS), numDirs);
-          timer.stop();
-        }
-        return true;
+        isModified = true;
+        break;
       }
     }
+    return logAndStopTimer(isModified, timer, numDirs);
+  }
+
+  private boolean logAndStopTimer(boolean isModified, Stopwatch timer, int numDirectories) {
     if (timer != null) {
-      logger.debug("No directories were modified. Took {} ms to check modification time of {} directories",
-          timer.elapsed(TimeUnit.MILLISECONDS), numDirs);
+      logger.debug("Directory {} was modified. Took {} ms to check modification time of {} directories",
+        isModified, timer.elapsed(TimeUnit.MILLISECONDS), numDirectories);
```

Comment:
It is actually useful to have a log message even for the case when no directories were modified..that way we can account for the elapsed time.  You can preserve the prior log message conditional based on the `isModified` flag.

Reply:
Thanks for catching this. Somehow I missed logging was done in both cases. Have made the change.